README.md

```diff
@@ -169,6 +171,51 @@ or its `constructor` object. The `of` method takes one argument:
 
     1. No parts of `b` should be checked
 
+### Foldable
+
+1. `u.reduce` is equivalent to `u.toArray().reduce`
+
+* `toArray`; derivable as `function() { return this.reduce(function(acc, x) { return acc.concat(x); }, []); }`
+
+#### `reduce` method
+
+A value which has a Foldable must provide a `reduce` method. The `reduce`
+method takes two arguments:
+
+    u.reduce(f, x)
+
+1. `f` must be a binary function
+
+    1. if `f` is not a function, the behaviour of `reduce` is unspecified.
+    2. The first argument to `f` must be the same type as `x`.
+    3. `f` must return a value of the same type as `x`
+
+1. `x` is the initial accumulator value for the reduction
+
+### Traversable
```

Comment:
Here's a gist of the laws: https://gist.github.com/DrBoolean/c4949a0f4b8f9ba8261f

A quick example would be:
```js
[Promise.of(1), Promise.of(2)].sequence(Promise.of)
//=> Promise([1,2])
```
It turns `[Promise]` to `Promise []`